app/src/lib/stores/updates/update-remote-url.ts

```diff
@@ -0,0 +1,51 @@
+import { Repository } from '../../../models/repository'
+import { IAPIRepository } from '../../api'
+import { GitStore } from '../git-store'
+import { urlMatchesRemote } from '../../repository-matching'
+import * as URL from 'url'
+
+export async function updateRemoteUrl(
+  gitStore: GitStore,
+  repository: Repository,
+  apiRepo: IAPIRepository
+): Promise<void> {
+  // I'm not sure when these early exit conditions would be met. But when they are
+  // we don't have enough information to continue so exit early!
+  if (!gitStore.defaultRemote) {
+    return
+  }
+  if (!repository.gitHubRepository) {
+    return
+  }
```

Comment:
if we want we could avoid this check by changing the type signature for this method to use `RepositoryWithGitHubRepository` instead of `Repository`. this may push this check up to the caller but i'm in favor of doing that for logic that is truly only for GitHub repositories.